clippy_lints/src/undocumented_unsafe_blocks.rs

```diff
@@ -711,3 +699,53 @@ fn text_has_safety_comment(src: &str, line_starts: &[RelativeBytePos], start_pos
         }
     }
 }
+
+macro_rules! item_like_kind_conversion {
+    ($varient:ident, $ty_convert_from:ty, $lft:lifetime) => {
+        impl<$lft> ::core::convert::From<$ty_convert_from> for ItemAlikeKind<$lft> {
+            fn from(value: $ty_convert_from) -> Self {
+                Self::$varient(value)
+            }
+        }
+    };
+}
+
+enum ItemAlikeKind<'hir> {
+    Item(ItemKind<'hir>),
+    TraitItem(hir::TraitItemKind<'hir>),
+    ImplItem(hir::ImplItemKind<'hir>),
+}
+
+item_like_kind_conversion!(Item, ItemKind<'hir>, 'hir);
+item_like_kind_conversion!(TraitItem, hir::TraitItemKind<'hir>, 'hir);
+item_like_kind_conversion!(ImplItem, hir::ImplItemKind<'hir>, 'hir);
+
+/// Representing the hir nodes that are item alike.
+///
+/// Note this does not includes [`Node::ForeignItem`] for now.
+struct ItemAlike<'hir> {
+    owner_id: hir::OwnerId,
+    kind: ItemAlikeKind<'hir>,
+    span: Span,
+}
+
+impl<'hir> ItemAlike<'hir> {
+    fn hir(node: &'hir Node<'hir>) -> Option<Self> {
+        let (owner_id, kind, span) = match node {
+            Node::Item(item) => (item.owner_id, item.kind.into(), item.span),
+            Node::TraitItem(ti) => (ti.owner_id, ti.kind.into(), ti.span),
+            Node::ImplItem(ii) => (ii.owner_id, ii.kind.into(), ii.span),
+            _ => return None,
+        };
+        Some(Self { owner_id, kind, span })
+    }
+
+    fn is_const_or_static(&self) -> bool {
+        matches!(
+            self.kind,
+            ItemAlikeKind::Item(ItemKind::Const(..) | ItemKind::Static(..))
+                | ItemAlikeKind::ImplItem(hir::ImplItemKind::Const(..))
+                | ItemAlikeKind::TraitItem(hir::TraitItemKind::Const(..))
+        )
+    }
```

Comment:
These could be two standalone functions without the new types/macros, something like

```rust
fn item_a_like(node: &Node) -> Option<(Span, HirId)> { ... }

fn is_const_or_static_item_a_like(node: &Node) -> bool { ... }
```

Also worth mentioning that you can use `.into()` to go from an `OwnerId` to a `HirId`

Reply:
Thanks, it becomes shorter too~

<sub>I guess I was thinking about putting it in `clippy_utils::higher`, but eventually realized this pattern is not that common (only one other lint, and a couple places in clippy_utils), so I just copy-pasted it back to this file lol. </sub>